src/mongocxx/uri.cpp

```diff
@@ -173,6 +173,10 @@ stdx::optional<bsoncxx::document::view> uri::credentials() {
     return bsoncxx::document::view(data, options_bson->len);
 }
 
+stdx::optional<std::int32_t> uri::srv_max_hosts() const {
+    return libmongoc::uri_get_option_as_int32(_impl->uri_t, MONGOC_URI_SRVMAXHOSTS, 0);
```

Comment:
```suggestion
    return _int32_option(_impl->uri_t, MONGOC_URI_SRVMAXHOSTS);
```

`mongoc_uri_get_option_as_int32` will return the fallback value (`0` in this case) if the option is not set. Using `_int32_option` will return `stdx::nullopt` if the option is not set.